ethdb/leveldb/leveldb.go

```diff
@@ -360,7 +355,14 @@ func (db *Database) meter(refresh time.Duration) {
 		db.nonlevel0CompGauge.Update(int64(stats.NonLevel0Comp))
 		db.seekCompGauge.Update(int64(stats.SeekComp))
 
-		// update tables amount
+		// leveldb maybe has more than 7 levels
+		if len(db.levelsGauge) != len(stats.LevelTablesCounts) {
+			db.levelsGauge = make([]metrics.Gauge, len(stats.LevelTablesCounts))
+			for i := range db.levelsGauge {
```

Comment:
Please only append the missing gauges to the end, instead of re-creating all of them.